server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/EmlRoutesTest.java

```diff
@@ -0,0 +1,128 @@
+        mailboxIdFactory = new InMemoryId.Factory();
+        mailboxManager = InMemoryIntegrationResources.defaultResources().getMailboxManager();
+
+        webAdminServer = WebAdminUtils
+            .createWebAdminServer(new EmlRoutes(new EmlService(mailboxManager, mailboxIdFactory)))
+            .start();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer)
+            .setBasePath(EmlRoutes.BASE_PATH)
+            .log(LogDetail.METHOD)
+            .build();
+    }
+
+    @AfterEach
+    public void stop() {
+        webAdminServer.destroy();
+    }
+
+    @Test
+    public void importEmlFileToMailboxShouldReturnNoContentWhenSuccess() throws MailboxException {
+
+        session = mailboxManager.createSystemSession("james@linagora.com");
+        dummyInbox = mailboxManager
+            .createMailbox(MailboxPath.forUser("james@linagora.com", "inbox"), session)
+            .get();
+
+        with()
+            .body(ClassLoader.getSystemResource("eml/mail.eml"))
+            .post("/" + dummyInbox.serialize() + "/messages")
+        .then()
+            .statusCode(HttpStatus.NO_CONTENT_204);
+    }
+
+    @Test
+    public void importEmlToMailBoxShouldAllowOtherFileExtensions() throws MailboxException {
+
+        session = mailboxManager.createSystemSession("james@linagora.com");
+        dummyInbox = mailboxManager
+            .createMailbox(MailboxPath.forUser("james@linagora.com", "inbox"), session)
+            .get();
+
+        with()
+            .body(ClassLoader.getSystemResource("eml/mail.json"))
+            .post("/" + dummyInbox.serialize() + "/messages")
+        .then()
+            .statusCode(HttpStatus.NO_CONTENT_204);
+    }
+
+    @Test
+    public void importEmlToMailboxShouldAllowEmptyEmlPath() throws MailboxException {
+
+        session = mailboxManager.createSystemSession("james@linagora.com");
+        dummyInbox = mailboxManager
+            .createMailbox(MailboxPath.forUser("james@linagora.com", "inbox"), session)
+            .get();
+
+        with()
+            .body(ClassLoader.getSystemResource(""))
+            .post("/" + dummyInbox.serialize() + "/messages")
+        .then()
+            .statusCode(HttpStatus.NO_CONTENT_204);
+    }
+
+    @Test
+    public void importEmlToMailboxShouldReturnBadRequestWhenUsernameDoesNotMatch() throws MailboxException {
+
+        session = mailboxManager.createSystemSession("EML");
+        dummyInbox = mailboxManager
+            .createMailbox(MailboxPath.forUser("EML", "inbox"), session)
+            .get();
+
+        with()
+            .body(ClassLoader.getSystemResource("eml/mail.eml"))
+            .post("/" + dummyInbox.serialize() + "/messages")
+        .then()
+            .statusCode(HttpStatus.BAD_REQUEST_400)
+            .body("message", is("Unable to create mailbox."));
+    }
+
+    @Test
+    public void importEmlToMailboxShouldReturnNotFoundWhenPathIsNotCorrect() throws MailboxException {
```

Comment:
again, the route doesn't care the file path...

Reply:
Not the file path, I mean the URL.